### PR TITLE
chore(plugin-proxy): deprecated warnings

### PR DESCRIPTION
**Pull request by SoonIter**
## Summary
add deprecated warning

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [x] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/cli/plugin-proxy/src/index.ts [hunk 1]
```diff
@@ -20,6 +21,9 @@ export const proxyPlugin = (): CliPlugin => {
         const rule = createProxyRule(internalDirectory, dev.proxy);
         if (!proxyServer) {
           proxyServer = new WhistleProxy({ port: 8899, rule });
+          logger.warn(
+            '[deprecated] @modern-js/plugin-proxy will no longer be maintained. Please consider migrating to other proxy tools.',
```
Comment (chenjiahan) [inline packages/cli/plugin-proxy/src/index.ts:25]:
```suggestion
            '[Deprecated] @modern-js/plugin-proxy is no longer maintained. Please consider migrating to other proxy tools, such as whistle.',
```

File: packages/document/main-doc/docs/en/components/global-proxy.mdx [hunk 2]
```diff
@@ -1,3 +1,7 @@
+:::warning
+`@modern-js/plugin-proxy` is no longer maintained and new versions will no longer be released.
```
Comment (chenjiahan) [inline packages/document/main-doc/docs/en/components/global-proxy.mdx:2]:
```suggestion
`@modern-js/plugin-proxy` is no longer maintained and new versions will no longer be released. Please consider migrating to other proxy tools, such as [whistle](https://github.com/avwo/whistle).
```

File: packages/document/main-doc/docs/zh/components/global-proxy.mdx [hunk 3]
```diff
@@ -1,3 +1,7 @@
+:::warning
+`@modern-js/plugin-proxy` 已不再维护，后续将不再发布新版本。
```
Comment (chenjiahan) [inline packages/document/main-doc/docs/zh/components/global-proxy.mdx:2]:
```suggestion
`@modern-js/plugin-proxy` 已不再维护，后续将不再发布新版本。请迁移到其他代理工具，比如 [whistle](https://github.com/avwo/whistle)。
```

**Review by chenjiahan — APPROVED** [review]

**Comment by SoonIter** [comment]
thanks~

**Comment by francecil** [comment]
Compared to using whistle directly, this tool has a higher level of integration with the project, including CA installation and starting/stopping whistle in accordance with the project lifecycle. What are the reasons for choosing to deprecate it?

**Pull request merged**